### PR TITLE
Adding docker inspect insights gathering

### PR DESCRIPTION
**Pull request by timclifford**
Running 'docker inspect' over each built image and adding these results to another insights configMap for later processing

### CODE REVIEW

**Review by bomoko — APPROVED** [review]

**Comment by tobybellwood** [comment]
I would be tempted here to consider using skopeo inspect to gather the image metadata - I think it provides similar coverage (ENVs, labels, layers), but we already use skopeo quite extensively in the build-deploy process as an alternative to docker

File: images/kubectl-build-deploy-dind/scripts/exec-generate-insights-configmap.sh [hunk 1]
```diff
@@ -1,13 +1,51 @@
 #!/bin/bash
 
-# SBOM config
 TMP_DIR="${TMP_DIR:-/tmp}"
 SBOM_OUTPUT="cyclonedx-json"
 SBOM_OUTPUT_FILE="${TMP_DIR}/${IMAGE_NAME}.cyclonedx.json.gz"
-SBOM_CONFIGMAP="lagoon-sbom-${IMAGE_NAME}"
+SBOM_CONFIGMAP="lagoon-insights-sbom-${IMAGE_NAME}"
+IMAGE_INSPECT_CONFIGMAP="lagoon-insights-image-inpsect-${IMAGE_NAME}"
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/scripts/exec-generate-insights-configmap.sh]:
```suggestion
IMAGE_INSPECT_CONFIGMAP="lagoon-insights-image-${IMAGE_NAME}"
```

Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/scripts/exec-generate-insights-configmap.sh:7]:
```suggestion
IMAGE_INSPECT_CONFIGMAP="lagoon-insights-image-${IMAGE_NAME}"
```

File: images/kubectl-build-deploy-dind/scripts/exec-generate-insights-configmap.sh [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+      kubectl --insecure-skip-tls-verify \
+          -n ${NAMESPACE} \
+          create configmap ${IMAGE_INSPECT_CONFIGMAP} \
+          --from-file=${IMAGE_INSPECT_OUTPUT_FILE}
+  fi
+  kubectl --insecure-skip-tls-verify \
+      -n ${NAMESPACE} \
+      label configmap ${IMAGE_INSPECT_CONFIGMAP} \
+      lagoon.sh/insightsProcessed- \
+      lagoon.sh/insightsType=image-inspect-gz \
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/scripts/exec-generate-insights-configmap.sh]:
```suggestion
      lagoon.sh/insightsType=image-gz \
```

**Comment by tobybellwood** [comment]
Couple of minor nits around naming - shortening image-inspect to *-image-*, but looks good and comes through OK in my testing

**Pull request merged**